### PR TITLE
Use consistent true/false flags in RTs; Update MOM6 to GFDL 20210806

### PR DESCRIPTION
**Pull request by DeniseWorthen**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Description

Provide a detailed description of what this PR does. What bug does it fix, or what feature does it add? Is a change of answers expected from this PR? Are any library updates included in this PR (modulefiles etc.)?

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes #733 

## Testing

Tested on gaea.intel; all baselines b4b

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [x] CI [https://github.com/ufs-community/ufs-weather-model/runs/3326981903]

## Dependencies

None

### CODE REVIEW

File: tests/default_vars.sh [hunk 1]
```diff
@@ -575,40 +575,40 @@ export FHMAX=${FHMAX:-`expr $DAYS \* 24`}
...
+export DO_SKEB=.false.
 export LNDP_TYPE=0
 export N_VAR_LNDP=0
-export LNDP_EACH_STEP=.F.
+export LNDP_EACH_STEP=.false.
 export SKEB=-999.
 export SPPT=-999.
 export SHUM=-999.
-export LNDP_VAR_LIST='XXX'
+export LNDP_VAR_LIST=XXX
```
Comment (climbfuji) [inline tests/default_vars.sh]: This is a list of strings, why no quotes around `XXX`? Seems that many other strings in `default_vars.sh` have quotes around them, but many not (ATMRES, for example). Should this be made consistent, too?

File: tests/parm/c96_HAFS_v0_hwrf.nml.IN [hunk 2]
```diff
@@ -122,7 +122,7 @@
   hwrf_samfshal  = .true.        ! HWRF SASshal
   asolfac_deep   = 0.89          ! HWRF SASdeep; GFS SAS:0.958
   asolfac_shal   = 0.89          ! HWRF SASdeep; GFS SAS:0.958
-  hurr_pbl       = .T.           ! HWRF moninedmf
+  hurr_pbl       = .true.           ! HWRF moninedmf
```
Comment (climbfuji) [inline tests/parm/c96_HAFS_v0_hwrf.nml.IN]: Please adjust whitespaces

File: tests/parm/esg_HAFS_v0_hwrf.nml.IN [hunk 3]
```diff
@@ -131,7 +131,7 @@
   hwrf_samfshal  = .true.        ! HWRF SASshal
   asolfac_deep   = 0.89          ! HWRF SASdeep; GFS SAS:0.958
   asolfac_shal   = 0.89          ! HWRF SASdeep; GFS SAS:0.958
-  hurr_pbl       = .T.           ! HWRF moninedmf
+  hurr_pbl       = .true.           ! HWRF moninedmf
```
Comment (climbfuji) [inline tests/parm/esg_HAFS_v0_hwrf.nml.IN]: Please adjust whitespaces

**Review by climbfuji — CHANGES_REQUESTED** [review]
Thanks for making these changes, tedious job.

I noted that the choice of using quotes around the values ini expressions like
```
export WARM_START='.false.'
```
are all over the place. No quotes, single quotes `'`, double quotes `"` are used. This is not a result of this PR, but something that could be cleaned up in a future PR.

**Comment by SMoorthi-emc** [comment]
Quotes are unnecessary

File: tests/parm/esg_HAFS_v0_hwrf.nml.IN (hunk 3 above)
Comment (DeniseWorthen) [reply tests/parm/esg_HAFS_v0_hwrf.nml.IN]: @climbfuji I don't understand the issue you pointed out about quotes in default_vars. Are you saying that there should be quotes? I tried to clean up default_vars for quotes, leaving only those where quotes separated values ("'atm' 'sfc'"). I didn't tackle quotes inside the tests yet.

I will fix the whitespaces

Comment (climbfuji) [reply tests/parm/esg_HAFS_v0_hwrf.nml.IN]: Sorry if my comments weren't clear, let's ignore the quotes for now. Thanks for fixing the whitespaces.

**Review by climbfuji — APPROVED** [review]

**Review by BrianCurtis-NOAA — APPROVED** [review]
Good to start RT

**Comment by DeniseWorthen** [comment]
@jiandewang can you start the wcoss dell and cray RTs? Thanks.

**Comment by jiandewang** [comment]
> @jiandewang can you start the wcoss dell and cray RTs? Thanks.

@DeniseWorthen will do right away

**Comment by jiandewang** [comment]
@DeniseWorthen jobs are running on Cray but I am having trouble to compile on P3.
Lmod has detected the following error: These module(s) exist but cannot be
loaded as requested: "yafyaml/v0.5.1", "mapl/v2.7.3"
   Try: "module spider yafyaml/v0.5.1 mapl/v2.7.3" to see how to load the
module(s).

Executing this command requires loading "yafyaml/v0.5.1" which failed while
processing the following module(s):

or you can see at /gpfs/dell2/ptmp/Jiande.Wang/FV3_RT/rt_39159/compile_001

**Comment by jiandewang** [comment]
@DeniseWorthen cray job done, I scp log to HERA at
/scratch2/NCEPDEV/climate/Jiande.Wang/For-others/For_Denise/UFS-PR-log/RegressionTests_wcoss_cray.log

I deleted the above log, will reload when new run is done

**Comment by BrianCurtis-NOAA** [comment]
Machine: gaea
Compiler: intel
Job: RT
Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/703937343/20210813153018/ufs-weather-model
Please manually delete: /lustre/f2/scratch/emc.nemspara/FV3_RT/rt_31643
Test hafs_regional_docn 086 failed failed 
Test hafs_regional_docn 086 failed in run_test failed 
Test hafs_regional_docn_oisst 087 failed failed 
Test hafs_regional_docn_oisst 087 failed in run_test failed 
Please make changes and add the following label back:
gaea-intel-RT

**Comment by DeniseWorthen** [comment]
@jiandewang We're killing all the auto-RT jobs because MOM6 is actually not pointing to your PR branch. That shouldn't explain your module load errors on Dell though.

**Comment by BrianCurtis-NOAA** [comment]
Machine: orion
Compiler: intel
Job: RT
Repo location: /work/noaa/nems/emc.nemspara/autort/pr/703937343/20210813103013/ufs-weather-model
Please manually delete: /work/noaa/stmp/bcurtis/stmp/bcurtis/FV3_RT/rt_252220
Test cpld_restart_bmark_v16 011 failed in run_test failed 
Please make changes and add the following label back:
orion-intel-RT

**Comment by BrianCurtis-NOAA** [comment]
Machine: cheyenne
Compiler: gnu
Job: RT
Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/703937343/20210813094509/ufs-weather-model
Please make changes and add the following label back:
cheyenne-gnu-RT

**Comment by BrianCurtis-NOAA** [comment]
Machine: cheyenne
Compiler: intel
Job: RT
Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/703937343/20210813093009/ufs-weather-model
Please make changes and add the following label back:
cheyenne-intel-RT

**Comment by BrianCurtis-NOAA** [comment]
Machine: hera
Compiler: intel
Job: RT
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/703937343/20210813153015/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_19974
Test compile 014 failed failed 
Test compile 019 failed failed 
Please make changes and add the following label back:
hera-intel-RT

**Comment by BrianCurtis-NOAA** [comment]
Machine: jet
Compiler: intel
Job: RT
Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/703937343/20210813153012/ufs-weather-model
Please manually delete: /lfs4/HFIP/h-nems/emc.nemspara/RT_RUNDIRS/emc.nemspara/FV3_RT/rt_168959
Test control_c192 024 failed failed 
Test control_2threads 018 failed failed 
Test control_c192 024 failed in run_test failed 
Test control_2threads 018 failed in run_test failed 
Test datm_bulk_cfsr 091 failed in run_test failed 
Test datm_bulk_gefs 092 failed in run_test failed 
Test fv3_esg_HAFS_v0_hwrf_thompson_debug 081 failed in run_test failed 
Test fv3_rrfs_v1alpha_debug 079 failed in run_test failed 
Test fv3_rrfs_v1beta_debug 078 failed in run_test failed 
Test cpld_control_wave 014 failed in run_test failed 
Test datm_mx025_cfsr 093 failed in run_test failed 
Test datm_mx025_gefs 094 failed in run_test failed 
Test datm_control_gefs 089 failed in run_test failed 
Test datm_control_cfsr 087 failed in run_test failed 
Test control_thompson_debug 072 failed in run_test failed 
Test fv3_HAFS_v0_hwrf_thompson_debug 080 failed in run_test failed 
Test datm_control_iau_gefs 090 failed in run_test failed 
Test control_thompson_extdiag_debug 074 failed in run_test failed 
Test fv3_gsd_debug 076 failed in run_test failed 
Test fv3_gsd_diag_debug 077 failed in run_test failed 
Test regional_control_debug 075 failed in run_test failed 
Test cpld_bmark_wave_v16_p7b 013 failed in run_test failed 
Test cpld_bmark_wave_v16 012 failed in run_test failed 
Test cpld_bmark_v16_nsst 011 failed in run_test failed 
Test cpld_restart_c384 008 failed in run_test failed 
Test cpld_restart_bmark_v16 010 failed in run_test failed 
Please make changes and add the following label back:
jet-intel-RT

**Comment by jiandewang** [comment]
@DeniseWorthen it is pointing to the right MOM branch
https://github.com/DeniseWorthen/ufs-weather-model/blob/feature/truefalse/.gitmodules#L33-L34
or did you just changed that ?

**Comment by DeniseWorthen** [comment]
The branch name is correct, but the hash points to the top of develop.

**Comment by jiandewang** [comment]
@DeniseWorthen now I see the problem, let me know when it's updated and I will re-do my wcoss run

**Comment by DeniseWorthen** [comment]
The hafs tests on gaea failed because they use ``WLCLK_dflt`` which is set to 15mins. Both runs timed out. The hafs DATM tests use WLCLK=30 explicitly.

**Comment by DeniseWorthen** [comment]
@DusanJovic-NOAA There seems to be a module load error on dell-p3, saying that netcdf-4.7.4 is not found.

**Comment by DeniseWorthen** [comment]
@jiandewang Please confirm MOM6 now points to the correct feature branch

**Comment by jiandewang** [comment]
@DeniseWorthen checked one minute ago, it is correct one now. Thanks.
So is it the right time for me to rerun on CRAY now ?

**Comment by DeniseWorthen** [comment]
@jiandewang Not yet. We are trying to get to the bottom of the module issue on Dell.

File: tests/default_vars.sh (hunk 1 above)
Comment (MinsukJi-NOAA) [reply tests/default_vars.sh]: Looks to me the single quotes may need to stay. For example:
```
control_ca:50:export LNDP_VAR_LIST="'XXX'"
control_ca_debug:44:export LNDP_VAR_LIST="'XXX'"
control_lndp:51:export LNDP_VAR_LIST="'smc' , 'vgf',"
control_lndp_debug:46:export LNDP_VAR_LIST="'smc' , 'vgf',"
```

**Review by MinsukJi-NOAA — APPROVED** [review]

**Comment by DeniseWorthen** [comment]
@MinsukJi-NOAA Thanks, I can re-enable quotes on the LNDP_VAR_LIST. I see now it does specifically call it a "list", which does imply it might have multiple items separated by spaces.

**Comment by DeniseWorthen** [comment]
@jiandewang could you please try on dell again? Since I updated MOM and changed one thing in default_vars, please be sure you try the updated PR.

**Comment by jiandewang** [comment]
> @jiandewang could you please try on dell again? Since I updated MOM and changed one thing in default_vars, please be sure you try the updated PR.

OK will try

**Comment by jiandewang** [comment]
> > @jiandewang could you please try on dell again? Since I updated MOM and changed one thing in default_vars, please be sure you try the updated PR.
> 
> OK will try

I think this fixed the issue, some of the compiling jobs have been done

**Comment by jiandewang** [comment]
@DeniseWorthen shall I start the full RT now ?

**Comment by DeniseWorthen** [comment]
@jiandewang Yes please. Thanks.

**Comment by jiandewang** [comment]
@DeniseWorthen  rerun on cray, see RT log on HERA at /scratch2/NCEPDEV/climate/Jiande.Wang/For-others/For_Denise/UFS-PR-log
  dell is still running at this moment

**Comment by jiandewang** [comment]
got disk quota issue on P3, re-running now

**Comment by BrianCurtis-NOAA** [comment]
Machine: jet
Compiler: intel
Job: RT
Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/703937343/20210813210012/ufs-weather-model
Please manually delete: /lfs4/HFIP/h-nems/emc.nemspara/RT_RUNDIRS/emc.nemspara/FV3_RT/rt_141404
Test hafs_regional_atm 082 failed in run_test failed 
Please make changes and add the following label back:
jet-intel-RT

**Comment by jiandewang** [comment]
RT jobs on dell P3 have failed 4 times due to disk quota

**Comment by BrianCurtis-NOAA** [comment]
@jiandewang what ACCNR are you using?

**Comment by jiandewang** [comment]
> @jiandewang what ACCNR are you using?

CFS-DEV

**Comment by jiandewang** [comment]
let me try to use stmp, ptmp is almost full

**Comment by BrianCurtis-NOAA** [comment]
@DeniseWorthen Jet was a slurm failure, i'll add the label for the RT again. Cheyenne was jobs sitting again, so i moved them to the premium queue and they finished. I didn't want them to get removed early AM tomorrow.
@jiandewang I will try with my ACCNR as well to see if it will finish OK. Will update in the AM if i don't hear from you beforehand

**Comment by jiandewang** [comment]
@DeniseWorthen  finally finished on dell, see RT log on HERA at /scratch2/NCEPDEV/climate/Jiande.Wang/For-others/For_Denise/UFS-PR-log

**Comment by climbfuji** [comment]
ufs-weather-model now points to hash f8a8e4c of MOM6, which is the head of the default branch. Seems ok.

**Pull request merged**